src/dockleHelper.ts

```diff
@@ -0,0 +1,94 @@
+import * as os from 'os';
+import * as path from 'path';
+import * as util from 'util';
+import * as fs from 'fs';
+import * as toolCache from '@actions/tool-cache';
+import * as core from '@actions/core';
+
+const stableDockleVersion = "0.2.4";
+const dockleLatestReleaseUrl = "https://api.github.com/repos/goodwithtech/dockle/releases/latest";
+const dockleToolName = "dockle";
+
+export async function getDockle(): Promise<string> {
+    const latestDockleVersion = await getLatestDockleVersion();
+    let cachedToolPath = toolCache.find(dockleToolName, latestDockleVersion);
+    if (!cachedToolPath) {
+        let dockleDownloadPath;
+        const dockleDownloadUrl = getDockleDownloadUrl(latestDockleVersion);
+        const dockleDownloadDir = `${process.env['GITHUB_WORKSPACE']}/_temp/tools/dockle`;
+        console.log(util.format("Could not find dockle in cache, downloading from %s", dockleDownloadUrl));
+
+        try {
+            dockleDownloadPath = await toolCache.downloadTool(dockleDownloadUrl, dockleDownloadDir);
+        } catch (error) {
+            throw new Error(util.format("Failed to download dockle from %s", dockleDownloadUrl));
+        }
+
+        fs.chmodSync(dockleDownloadPath, "777");
+        const untarredDocklePath = await toolCache.extractTar(dockleDownloadPath);
+        cachedToolPath = await toolCache.cacheDir(untarredDocklePath, dockleToolName, latestDockleVersion);
+    }
+
+    const dockleToolPath = findDockle(cachedToolPath);
+    fs.chmodSync(dockleToolPath, "777");
+
+    return dockleToolPath;
+}
+
+async function getLatestDockleVersion(): Promise<string> {
+    return toolCache.downloadTool(dockleLatestReleaseUrl).then((downloadPath) => {
+        const response = JSON.parse(fs.readFileSync(downloadPath, 'utf8').toString().trim());
+        if (!response.tag_name) {
+            return stableDockleVersion;
+        }
+
+        return response.tag_name.substring(1);

```

Comment:
low priority: Use semver.clean() instead? (Thinking of a case when they start returning version like 2.0.0 instead of v2.0.0. This would be more robust.)

Reply:
Okay. I'll replace this with semver.clean